agent/dns_test.go

```diff
@@ -545,6 +545,7 @@ func TestDNS_NodeLookup_CNAME(t *testing.T) {
 
 	m := new(dns.Msg)
 	m.SetQuestion("google.node.consul.", dns.TypeANY)
+	m.SetEdns0(8192, true)
```

Comment:
This test started to fail because we are now trimming every response instead of only some of them. So I had to `setEdns0` to increase the size of the response.

I guess this is probably expected behaviour, and is now more correct? I'm honestly not sure.

Reply:
The spec state that without eDNS we should trim to 512 by default and this is what we do now, So I think we are good